src/main/java/seedu/address/model/student/Training.java

```diff
@@ -57,10 +57,11 @@ public void addStudent(Student student) {
      * @param student
      */
     public void removeStudent(Student student) {
-        if (students.contains(student)) {
-            throw new DuplicateStudentException();
-        } else {
-            this.students.add(student);
+        Set<Student> studentsCopy = new HashSet<>(students);
```

Comment:
Might need to remove/add the LocalDateTime from/to the student's training schedule treeSet container as well, feel free to use the methods alr implemented in student class for this, so that student training schedule panel will update too